src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -42,7 +42,13 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
         CommandResult commandResult;
-        Command command = addressBookParser.parseCommand(commandText);
+        Command command;
+        String selectedDeckName = this.model.getSelectedDeckName();
```

Comment:
Instead of getting the selectedDeckName here, do you want to bring in to the Command::execute method? In terms of OOP, I dont think a parser needs to pass around the selectedDeckName? I may be wrong and misunderstood your design choices though!